lms/djangoapps/learner_dashboard/views.py

```diff
@@ -11,39 +7,30 @@
 from django.http import Http404
 
 from edxmako.shortcuts import render_to_response
-from openedx.core.djangoapps.programs.utils import get_user_enrolled_programs
+from openedx.core.djangoapps.programs.utils import get_engaged_programs
 from openedx.core.djangoapps.programs.models import ProgramsApiConfig
 from student.views import get_course_enrollments
 
 
-log = logging.getLogger(__name__)
-
-
-def _get_xseries_url():
-    '''Create the xseries advertising link url'''
-    xseries_url = None
-    if ProgramsApiConfig.current().show_xseries_ad:
-        xseries_url = urljoin(settings.MKTG_URLS.get('ROOT'), 'xseries')
-    return xseries_url
-
-
 @login_required
 @require_GET
 def view_programs(request):
-    '''
-    This is the view to see all the programs the learner has been enrolled in
-    '''
+    """View programs in which the user is engaged."""
     if not ProgramsApiConfig.current().is_student_dashboard_enabled:
-        raise Http404("learner dashboard not enabled")
+        raise Http404('Program listing page not enabled.')
 
-    course_enrollments = list(get_course_enrollments(request.user, None, []))
+    enrollments = list(get_course_enrollments(user, None, []))
+    programs = get_engaged_programs(request.user, enrollments)
 
-    programs = get_user_enrolled_programs(
-        request.user,
-        [enrollment.course_id for enrollment in course_enrollments]
-    )
+    # TODO: Pull 'xseries' string from configuration model.
```

Comment:
To support open source site operators, we should try to keep hardcoded references to XSeries out of this code. We should be able to use config models to supply this view with marketing-related information. I've intentionally not done that here to keep this set of changes small.